control-plane/pkg/reconciler/consumergroup/consumergroup.go

```diff
@@ -845,3 +848,51 @@ func metricTagsOf(ctx context.Context, cg *kafkainternals.ConsumerGroup) (contex
 func keyOf(cg metav1.Object) string {
 	return types.NamespacedName{Namespace: cg.GetNamespace(), Name: cg.GetName()}.String()
 }
+
+// deleteKedaObjects is responsible for deleting Keda (Kubernetes-based Event Driven Autoscaling) objects, "scaled
+// objects" and "trigger authentications" from all available namespaces.
+//
+// When Keda creates a scaled object, it also generates a Horizontal Pod Autoscaler. This resource will also be removed
+// when the scaled object is deleted.
+func (r *Reconciler) deleteKedaObjects(ctx context.Context) error {
```

Comment:
It looks like this will delete all KEDA scaled objects and trigger authentications in the cluster, not just those created by the kafka-controller. Maybe we could leverage the fact that the name of the scaledobject (in keda.GenerateScaledObject) and trigger authentication (in keda.GenerateTriggerAuthentication) for each consumergroup is deterministic and we can generate it each time?

Then, since the `ReconcileKind` function will be called once per consumergroup when we update the configmap, we could:
1. Get the name of the scaledobject associated with the consumergroup and then delete it
2. Get the name of the trigger authentication associated with the consumergroup and then delete it

This way, we won't delete any extra KEDA resourced

Reply:
that sounds like a good idea. I tried to follow that on my last commit.